core/trino-spi/src/main/java/io/trino/spi/connector/EntityKindAndName.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import static java.util.Objects.requireNonNull;
+
+public class EntityKindAndName
+{
+    private final String entityKind;
+    // TODO: Should this instead be List<String> like QualifiedName?
```

Comment:
Yes, but it depends on what we're trying to achieve. Flattening a multipart QualifiedName into a string is problematic due to lack of escaping of special characters. For instance a name such as "a.b".c.d cannot be flattened to a string "a.b.c.d" without loss of information. If we want to be able to recover the catalog, schema and name, we need to keep them separate.

In fact. We may just want to separate catalog/schema/name as different elements.

Reply:
Let me know what you prefer, @martint, and I'll change it to conform.

Reply:
I decided that `EntityKindAndName` should use `List<String>` as the name, and made that change.

I also added `SystemSecurityMetadata.getAllEntityKindPrivileges` to provide a way to know at parse time if a privilege is legal.

Finally, I created [this galaxy-trino branch](https://github.com/starburstdata/galaxy-trino/tree/djs/exotic-privileges) to support the new Trino entity privilege APIs once they are merged.